### PR TITLE
[kube-prometheus-stack] Fix etcd dashboards

### LINKED ISSUE #3889: [kube-prometheus-stack] Grafana dashboard for etcd broken
### Describe the bug a clear and concise description of what the bug is.

Recent commit bdfef97dd9b62afc3a59bd412b80cdb9f3e2a791 broke the etcd Grafana dashboard.

The problem is that a variable that was previously named "cluster" was renamed to a name of "job" but there are still ocurrences of $cluster in the panels.

### What's your helm version?

irrelevant

### What's your kubectl version?

irrelevant

### Which chart?

kube-prometheus-stack

### What's the chart version?

51.6.0

### What happened?

The Grafana dashboard for etcd has no data at all, as the not defined variable $cluster is used.

### What you expected to happen?

The Grafana dashboard should work.

### How to reproduce it?

Install the Grafana dashboards for etcd.

### Enter the changed values of values.yaml?

grafana:
   forceDeployDashboards: true

### Enter the command that you execute and failing/misfunctioning.

not applicable

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by QuentinBisson**
#### What this PR does / why we need it

This PR regenerates the etcd dashboard as the mixins were fixed as a result of discussions from https://github.com/prometheus-community/helm-charts/pull/3880#issuecomment-1765267105

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #3889

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by QuentinBisson** [comment]
@GMartinez-Sisti could you look at this one?

File: charts/kube-prometheus-stack/hack/sync_grafana_dashboards.py [hunk 1]
```diff
@@ -58,8 +58,8 @@ def new_representer(dumper, data):
     'scheduler': ' .Values.kubeScheduler.enabled',
     'node-rsrc-use': ' .Values.nodeExporter.enabled',
     'node-cluster-rsrc-use': ' .Values.nodeExporter.enabled',
-    'nodes': ' .Values.nodeExporter.operatingSystems.linux.enabled',
-    'nodes-darwin': ' .Values.nodeExporter.operatingSystems.darwin.enabled',
+    'nodes': ' (and .Values.nodeExporter.enabled .Values.nodeExporter.operatingSystems.linux.enabled)',
+    'nodes-darwin': ' (and .Values.nodeExporter.enabled .Values.nodeExporter.operatingSystems.darwin.enabled)',
```
Comment (GMartinez-Sisti) [inline charts/kube-prometheus-stack/hack/sync_grafana_dashboards.py:62]: @QuentinBisson are these intentional for this commit? Looks like they are not related with the issue, from what I understood of the issue.

Comment (QuentinBisson) [reply charts/kube-prometheus-stack/hack/sync_grafana_dashboards.py:62]: Yes sorry, those are intentional as we removed `.Values.nodeExporter.enabled` as a condition in a previous PR while it does not make sense to have `.Values.nodeExporter.operatingSystems.linux.enabled` generate dashboard when node exporter itself is missing

Comment (GMartinez-Sisti) [reply charts/kube-prometheus-stack/hack/sync_grafana_dashboards.py:62]: Alright, just checking 😄

**Review by GMartinez-Sisti — APPROVED** [review]

**Comment by QuentinBisson** [comment]
Thank you :)

**Comment by jkroepke** [comment]
Thanks!

**Pull request merged**